packages/common-utils/src/core/renderChartConfig.ts

```diff
@@ -2,10 +2,20 @@ import isPlainObject from 'lodash/isPlainObject';
 import * as SQLParser from 'node-sql-parser';
 import SqlString from 'sqlstring';
 
-import { ChSql, chSql, concatChSql, wrapChSqlIfNotEmpty } from '@/clickhouse';
+import {
+  ChSql,
+  chSql,
+  ColumnMeta,
+  concatChSql,
+  wrapChSqlIfNotEmpty,
+} from '@/clickhouse';
 import { Metadata } from '@/core/metadata';
 import { CustomSchemaSQLSerializerV2, SearchQueryBuilder } from '@/queryParser';
 
+const MAP_CONTAINS_OPTIMIZATION_ENABLED =
+  process.env.NEXT_PUBLIC_MAP_CONTAINS_OPTIMIZATION_ENABLED ||
+  process.env.NODE_ENV !== 'production';
```

Comment:
I'd suggest removing this (in case somehow NODE_ENV is not set properly in prod) and add `NEXT_PUBLIC_MAP_CONTAINS_OPTIMIZATION_ENABLED` to your `.env.local` instead